IonDotnet/Tree/Impl/IonSequence.cs

```diff
@@ -87,12 +84,9 @@ public override bool Remove(IIonValue item)
         {
             ThrowIfLocked();
             ThrowIfNull();
-            if (NullFlagOn() || item?.Container != this)
```

Comment:
Do we still need the `NullFlagOn()` check?

Reply:
```NullFlagOn()``` is in IonValue and is used by different classes for different checks. For example ```IsEquivalentTo```

Reply:
Sure. What I mean is that the removal of the `NullFlagOn()` check at this location doesn't seem related to the rest of the CR. Was that check never needed here, or is it related in some way that I don't understand?

Reply:
I see, sorry I understood it wrong.
Seems like I shouldn't have removed it.